### PR TITLE
coveralls + testing and contributing doc

### PR DESCRIPTION
**Pull request by DylannCordel**
PR to enhance the code quality of pybbm. Currently, it's difficult to find dead code or untested lines. With [coveralls](https://pypi.python.org/pypi/coveralls/), you can have statistics and identify untested lines for python files. This PR:

* adds coveralls to tox and travis config files and coveralls badge in the README. (hovel organisation must sign-in for a free account on coveralls). From webu:features/coveralls branch, this badge looks like this: [Coverage Status](https://coveralls.io/github/webu/pybbm?branch=features%2Fcoveralls)
* adds some `pragma: no cache` comments for lines which are not relevent to the code quality (deprecated code for exemple)
* enhances the documentation about testing and add a specific contributing page.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
     package_data={'': ['pybb/templates', 'pybb/static']},
     zip_safe=False,
     install_requires=[
+        'Django>=1.8',
```
Comment (skolsuper) [inline setup.py]: This change probably doesn't belong in this PR

File: tox.ini [hunk 2]
```diff
@@ -1,5 +1,17 @@
 [tox]
-envlist = py{27,34,35}-django1{8,9,10}
+envlist = py{27,34,35}-django1{8,9,10} coverage
+
+[testenv:coverage]
+basepython = python2.7
+deps =
+    Django>=1.8,<1.9
+    -Urtest/test_project/requirements_test.txt
```
Comment (skolsuper) [inline tox.ini]: is there a space missing here?

File: pybb/templatetags/pybb_tags.py [hunk 3]
```diff
@@ -84,7 +84,7 @@ def pybb_link(object, anchor=''):
 
 
 @register.filter
-def pybb_topic_moderated_by(topic, user):
+def pybb_topic_moderated_by(topic, user):  # pragama: no cover
```
Comment (skolsuper) [inline pybb/templatetags/pybb_tags.py]: typo pragama -> pragma

File: .travis.yml [hunk 4]
```diff
@@ -19,7 +19,9 @@ before_script:
 install:
   - if [[ $DB == mysql ]]; then pip install -qU mysqlclient; fi
   - if [[ $DB == postgres ]]; then pip install -qU psycopg2; fi
-  - pip install -qU $DJANGO
+  - pip install -qU $DJANGO coveralls
```
Comment (skolsuper) [inline .travis.yml]: maybe should add to `test_requirements.txt` instead?

File: .travis.yml [hunk 5]
```diff
@@ -19,7 +19,9 @@ before_script:
...
   - CFLAGS="-O0" pip install -qU -r test/test_project/requirements_test.txt
 script:
-  - python setup.py test
+  - coverage run setup.py test
```
Comment (skolsuper) [inline .travis.yml]: I vaguely remember another python oss project I have seen ran coverage only on one combination of the matrix, as it slows down the tests significantly and results from only one run are counted anyway

File: tox.ini (hunk 2 above)
Comment (DylannCordel) [inline tox.ini]: No it's not a typo : I kept the same syntax which was used in this file. I have seen this in some projects. I suppose it's for some version of tox or OS ?

File: .travis.yml (hunk 4 above)
Comment (DylannCordel) [inline .travis.yml]: IMHO, it's better if we leave it outside test_requirements because it's not required to run tests. Only required when we want a coverage run (tox -e coverage)

File: .travis.yml (hunk 5 above)
Comment (DylannCordel) [inline .travis.yml]: After an epic battle with a `;`, it's coervalls il only performed for django1.8 + postgres :-). Thanks for the tip.

File: setup.py (hunk 1 above)
Comment (DylannCordel) [inline setup.py]: fixed. Thx. I'll create another PR for this change. PyBB requires Django>=1.8 so it should be in the setup.py.

**Comment by DylannCordel** [comment]
Thanks for your feedback @skolsuper  
I push another commit to fix typos and run coveralls only for django1.8 + postgres.

**Comment by lampslave** [comment]
>hovel organisation must sign-in for a free account on coveralls

@zeus, what do you think about it? https://coveralls.io/
